src/main/java/Poppy.java

```diff
@@ -0,0 +1,155 @@
+                break;
+            case "List":
+                showList(list);
+                break;
+            case "todo":
+                ToDo task= new ToDo(str[1]);
+                list[counter] = task;
+                counter++;
+                break;
+            case "deadline":
+                String[] deadlinestring = str[1].split( "/by", 2);
+                Deadline deadline = new Deadline(deadlinestring[0], deadlinestring[1]);
+                list[counter] = deadline;
+                counter++;
+                break;
+            case "event":
+                String[] eventstring = str[1].split( "/from", 2);
+                Events event = new Events(eventstring[0], eventstring[1]);
+                list[counter] = event;
+                counter++;
+                break;
+            default:
+                list[counter] = new Task(input);
+                counter++;
+                echo(list[counter - 1]);
+            }
+            input =sc.nextLine();
+        }
+        System.out.println("Bye. Hope to see you again soon!");
+        System.exit(0);
+    }
+
+    public static void echo(Task task) {
+        System.out.println("added: " + task.description);
+    }
+
+    public static void markAsDone(Task[] list, String[] str) {
+        int index = Integer.parseInt(str[1]) - 1;
+        if (list[index]== null){
+            System.out.println("Sorry, list number does not exist");
+        }else {
+            list[index].markAsDone();
+            System.out.println("Nice! I've marked this task as done:");
+            System.out.println(list[index].taskType() + " [" + list[index].getStatusIcon() + "] " + list[index].description);
+        }
+    }
+
+    public static void showList(Task[] list){
+        System.out.println("Here are the tasks in your list: ");
+        for (int i = 0; i < list.length; i++) {
+            if (list[i] != null) {
+                System.out.println((i + 1) + ". " +list[i].taskType() +" [" + list[i].getStatusIcon() + "] " + list[i].description);
+            }
+        }
+    }
+
+    public static void markAsNotDone(Task[] list, String[] str) {
+        int index = Integer.parseInt(str[1])- 1;
+        if (list[index]== null){
+            System.out.println("Sorry, list number does not exist");
+        }else {
+            list[index].markAsNotDone();
+            System.out.println("OK, I've marked this task as not done yet:");
+            System.out.println(list[index].taskType()+ " [" + list[index].getStatusIcon() + "] " + list[index].description);
+        }
+    }
+
+    public static class Task {
+        protected String description;
+        protected boolean isDone;
+
+        public Task(String description) {
+            this.description = description;
+            this.isDone = false;
+        }
+
+        public String getStatusIcon() {
+            return (isDone ? "X" : " "); // mark done task with X
+        }
+
+        public void markAsDone() {
```

Comment:
using "setDone" is better for function naming. for IP i think is ok to do this since its individual work anyways, but when we doing tP, it would be good to take note for this since we want the code to look like its written by one person.